src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -191,18 +194,13 @@ private Task StartAsyncCore()
                     _httpClient?.Dispose();
 
                     _logger.RaiseClosed(_connectionId);
-
-                    var closedEventHandler = Closed;
-                    if (closedEventHandler != null)
+                    if (t.IsFaulted)
```

Comment:
Do we have a test for this?

Reply:
Some of the EndToEnd tests hit this path with a faulted task. 
This is where we had the issue with the wrapped faulted task